### PR TITLE
feat: upgrade GWAPI to v0.6.0-rc1

### PR DESCRIPTION
**Pull request by Xunzhuo**
Fixes: https://github.com/envoyproxy/gateway/issues/716

Signed-off-by: bitliu <bitliu@tencent.com>

### CODE REVIEW

**Comment by arkodg** [comment]
hey @Xunzhuo looking at https://github.com/kubernetes-sigs/gateway-api/pull/1499 & https://github.com/projectcontour/contour/pull/4849 we might need to update the `Ready` Listener condition with `Programmed`

**Comment by Xunzhuo** [comment]
Cool, thanks @arkodg, I will update it later.

**Comment by youngnick** [comment]
Yes, the two biggest changes are `Ready` -> `Programmed`, and the move of everything to `Accepted` for indicating attachment/accepted.

**Comment by arkodg** [comment]
thanks @youngnick 
@Xunzhuo this is the relevant PR reg `Accepted` condition https://github.com/kubernetes-sigs/gateway-api/pull/1447

File: internal/gatewayapi/contexts_test.go [hunk 1]
```diff
@@ -35,12 +35,12 @@ func TestContexts(t *testing.T) {
 	lctx := gctx.GetListenerContext("http")
 	require.NotNil(t, lctx)
 
-	lctx.SetCondition(v1beta1.ListenerConditionDetached, metav1.ConditionTrue, v1beta1.ListenerReasonUnsupportedProtocol, "HTTPS protocol is not supported yet")
+	lctx.SetCondition(v1beta1.ListenerConditionAccepted, metav1.ConditionTrue, v1beta1.ListenerReasonUnsupportedProtocol, "HTTPS protocol is not supported yet")
```
[the remark this answers is not in the document]
Comment (Xunzhuo) [reply internal/gatewayapi/contexts_test.go]: True, updated, thanks!

**Review by arkodg — APPROVED** [review]
❤️

**Pull request merged**